### PR TITLE
ILMHistoryStore bulk size tweak

### PR DESCRIPTION
**Pull request by joegallo**
The crux of the changes are in `ILMHistoryStore`.

The most interesting test here can be run with this command:

```
./gradlew --info ':x-pack:plugin:ilm:internalClusterTest' --tests "org.elasticsearch.xpack.ilm.ILMHistoryStoreIT.testPutAsyncStressTest"
```

Related to #68468 (but I would not say it closes that issue)

### CODE REVIEW

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStore.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ public void putAsync(ILMHistoryItem item) {
             // TODO: remove the threadpool wrapping when the .add call is non-blocking
             // (it can currently execute the bulk request occasionally)
             // see: https://github.com/elastic/elasticsearch/issues/50440
-            threadPool.executor(ThreadPool.Names.GENERIC).execute(() -> {
+            threadPool.executor(ThreadPool.Names.MANAGEMENT).execute(() -> {
```
Comment (original-brownbear) [inline x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStore.java:153]: I thought about this some more and looked at some recent SDHs. We already have a bit of an issue with slow `MANAGEMENT` threads from stats.
I think, even though this is a theoretically more correct. fix as we discussed, this has quite some potential to end weird (thinking about submitting 10k of these tasks in a hot loop and 5 of them blocking the management pool forever if the data node is slow for some reason).
-> I'd say, lets just stick with generic here and assume that bulk size limit increase will be all that's needed.
Maybe we can go for a configurable value here via an undocumented system property just so we have an out n case this causes trouble or is not enough in some corner case?

**Review by original-brownbear — COMMENTED** [review]
Thanks Joe! Mainly just a comment on the thread-pool to use here now that I had some time to think this through and on how we can make this even safer :)

File: x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStore.java (hunk 1 above)
Comment (joegallo) [reply x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStore.java]: 2574688 and aa5be68

**Comment by joegallo** [comment]
Converted this PR back to a draft because I don't want to introduce a new test and awaitsfix it on the same PR, also the history here was getting out of hand and this was always meant as 'just' a WIP PR. I've spun https://github.com/elastic/elasticsearch/pull/80902 out of this, and that's a very reasonable (and reviewable) subset of the work that I've one here.

**Comment by joegallo** [comment]
Closing in favor of #91238

**Pull request closed without merging**